samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -64,12 +64,13 @@
   private volatile String ephemeralPath = null;
   private final ZkKeyBuilder keyBuilder;
   private final int connectionTimeoutMs;
-  private final String processorId = "TO BE PASSED IN THE CONSTRUCTOR"; //TODO
+  private final String processorId;
 
-  public ZkUtils(ZkKeyBuilder zkKeyBuilder, ZkClient zkClient, int connectionTimeoutMs) {
+  public ZkUtils(String processorId, ZkKeyBuilder zkKeyBuilder, ZkClient zkClient, int connectionTimeoutMs) {
```

Comment:
I prefer to keep this class as a util class for interacting with Zk and hence, avoid any state associated with it. If processorId is only used in log lines, I would rather keep it out.